test/ibmq/test_ibmq_job.py

```diff
@@ -279,16 +305,37 @@ def test_retrieve_job_uses_appropriate_backend(self, provider):
                               real_backend.retrieve_job, job_sim.job_id())
         self.assertIn('belongs to', str(context_manager.warning))
 
+    def test_retrieve_job_error_backend(self, provider):
+        """Test retrieving an invalid job from a backend."""
+        backends = provider.backends(simulator=False)
+        backend = least_busy(backends)
+
+        self.assertRaises(IBMQBackendError, backend.retrieve_job, 'BAD_JOB_ID')
+
     @requires_provider
-    def test_retrieve_job_error(self, provider):
-        """Test retrieving an invalid job."""
+    def test_retrieve_job_error_backend_service(self, provider):
+        """Test retrieving an invalid job from backend service."""
         self.assertRaises(IBMQBackendError, provider.backends.retrieve_job, 'BAD_JOB_ID')
 
-    @requires_provider
-    def test_get_jobs_filter_job_status(self, provider):
+    def test_get_jobs_filter_job_status_backend(self, provider):
         """Test retrieving jobs from a backend filtered by status."""
         backends = provider.backends(simulator=False)
         backend = least_busy(backends)
+        with warnings.catch_warnings():
+            # Disable warnings from pre-qobj jobs.
+            warnings.filterwarnings('ignore',
```

Comment:
Can you check if we can remove this line (making the `with` block not needed)?

Reply:
I didn't think they would be necessary as I was reintroducing the tests. But, I left them thinking they may have been for something else. I didn't notice any warnings being caught while running the tests. And, the only reason a warning would be triggered now in `IBMQBackend` is if one tries to do `IBMQBackend.retrieve_job` with an inappropriate backend. So, I went ahead and removed it, because it doesn't seem to be in use.

Reply:
Sounds good!